cmd/querier/main.go

```diff
@@ -88,6 +88,7 @@ func main() {
 
 	subrouter := server.HTTP.PathPrefix("/api/prom").Subrouter()
 	subrouter.PathPrefix("/api/v1").Handler(middleware.AuthenticateUser.Wrap(promRouter))
+	subrouter.Path("/remote_read").Handler(middleware.AuthenticateUser.Wrap(http.HandlerFunc(queryable.Q.RemoteReadHandler)))
```

Comment:
The handler in the distributor is called `push`, so this could be `pull` :-P

Also, you'll need to add the right route to authfe so this gets through - let me know if you need a pointer.